llvm/lib/Target/RISCV/RISCVISelLowering.cpp

```diff
@@ -18370,6 +18370,10 @@ RISCVTargetLowering::getRegForInlineAsmConstraint(const TargetRegisterInfo *TRI,
       // TODO: Support fixed vectors up to XLen for P extension?
       if (VT.isVector())
         break;
+      if (VT == MVT::f32 && Subtarget.hasStdExtZfinx())
```

Comment:
Is Zhinx broken too?

Reply:
That's a good question. I will try it out later to see.

Reply:
The simple test case for f16 and f32 doesn't cause problems because the necessary physreg copies exist, but I think it's still useful to use the right register class for the respective value.